tee-worker/core/rpc-client/src/direct_client.rs

```diff
@@ -59,6 +60,7 @@ pub trait DirectApi {
 	fn send(&self, request: &str) -> Result<()>;
 	/// Close any open websocket connection.
 	fn close(&self) -> Result<()>;
+	fn get_next_nonce(&self, shard: &ShardIdentifier, account: &AccountId) -> Result<String>;
```

Comment:
Why not return `u32` or `Index` (whatever the nonce type is)?

And can yo move this fn below `get_state_metadata_raw` so that it belongs to Litentry add-ups?